unix_integration/resolver/src/resolver.rs

```diff
@@ -711,8 +722,26 @@ impl Resolver {
     }
 
     async fn get_nssgroup(&self, grp_id: Id) -> Result<Option<NssGroup>, ()> {
-        if let Some(nss_group) = self.system_provider.get_nssgroup(&grp_id).await {
+        if let Some(mut nss_group) = self.system_provider.get_nssgroup(&grp_id).await {
             debug!("system provider satisfied request");
+
+            for client in self.clients.iter() {
+                if let Some(extend_group_id) = client.has_map_group(&nss_group.name) {
+                    let token = self.get_grouptoken(extend_group_id.clone()).await?;
```

Comment:
Why does this one not use the cached token?

Reply:
get_nssgroups is enumeration of everything on the system, so it smashes the cache. It's generally only used during "getent group" so we only want to shows what's locally available.

Where as get_nssgroup (singular) is for getting the defaults of a singular named group, so we want to call the backend to check if it's up-to-date or not.